### PR TITLE
Optimise one

### PR DESCRIPTION
**Pull request by pbiology**
## Optimisation round 1

This PR is a small update with various improvements. No specific functionality has been added, but rather some small tweaks to make the core easier to understand and more streamlined.

### Added
- Fusion_inspect only runs on a filtered list of detected fusions. Prevents pizzly from adding thousands of fusions
- Moved logic of how threads were started to separate file
- Separated logs for rnaseq and rnafusion
- Improvements to logging
- Added possibility of specifying aligner in the config
- Nextflow config file now in repo. Might change in future to use an institutional config directly in nf-core.
- Black lining

### To test
```
git clone git@github.com:ClinicalGenomicsGBG/qd-rnaseq-wrapper.git
cd qd-rnaseq-wrapper
git checkout optimise_one
#(optional step if no conda env built)
conda env create -f conda-env/qd-rnaseq.yaml -p conda-env/qd-rnaseq
conda activate conda-env/qd-rnaseq
python wrapper.py --fastqdir /home/xlinak/inbox/test_squid/data --outdir ./test_outdir --testrun
```

### CODE REVIEW

File: tools/helpers.py [hunk 1]
```diff
@@ -87,18 +89,23 @@ def sanitize_fastqdir(fastqdir: str):
...
 def build_rnaseq_command(
-    config, outdir: str, logdir: str, ss_path: str, testrun=False, save_reference=False
-) -> list:
+    config,
+    outdir: str,
+    logdir: str,
+    ss_path: str,
+    testrun=False,
+    save_reference=False,
+) -> dict:
```
Comment (dodslaser) [inline tools/helpers.py]: The return value could be more strictly typed as `dict[str, list[str]]`

File: tools/helpers.py [hunk 2]
```diff
@@ -148,18 +175,25 @@ def build_rnaseq_command(
...
+
 
-def build_rnafusion_command(config, outdir: str, logdir: str, ss_path: str, testrun=False) -> list:
+def build_rnafusion_command(
+    config,
+    outdir: str,
+    logdir: str,
+    ss_path: str,
+    testrun=False,
+) -> dict:
```
Comment (dodslaser) [inline tools/helpers.py]: The return value could be more strictly typed as `dict[str, list[str]]`

File: tools/helpers.py [hunk 3]
```diff
@@ -205,4 +259,30 @@ def build_rnafusion_command(config, outdir: str, logdir: str, ss_path: str, test
     rnafusion_command.append("--outdir")
     rnafusion_command.append(os.path.join(outdir, "rnafusion"))
 
-    return rnafusion_command
+    return {"nf-core/rnafusion": rnafusion_command}
+
+
+def start_pipe_threads(pipe_dict: dict, logger):
```
Comment (dodslaser) [inline tools/helpers.py]: Functions without an explicit `return` could be type hinted as returning `None`

eg. `def start_pipe_threads(pipe_dict: dict, logger) -> None:` ...

File: wrapper.py [hunk 4]
```diff
@@ -2,33 +2,63 @@
...
+    help="Skip the nf-core/rnafusion pipeline",
+)
+@click.option(
+    "--save-reference",
+    is_flag=True,
+    help="Save the genome references in outfolder",
+)
+def main(
+    fastqdir, outdir, strandedness, testrun, skip_rnaseq, skip_rnafusion, save_reference
+):
```
Comment (dodslaser) [inline wrapper.py]: Missing type hints, and line wrapping to match other functions

**Review by dodslaser — APPROVED** [review]

File: tools/helpers.py (hunk 1 above)
Comment (pbiology) [reply tools/helpers.py]: Fixed in be54ef5

File: tools/helpers.py (hunk 2 above)
Comment (pbiology) [reply tools/helpers.py]: Fixed in be54ef5

File: tools/helpers.py (hunk 3 above)
Comment (pbiology) [reply tools/helpers.py]: Fixed in be54ef5

File: wrapper.py (hunk 4 above)
Comment (pbiology) [reply wrapper.py]: Yes, agreed!
Fixed in be54ef5

**Comment by pbiology** [comment]
Did you also try to testrun this @dodslaser?

**Review by dodslaser — APPROVED** [review]

**Comment by dodslaser** [comment]
> Did you also try to testrun this @dodslaser?

The included test fails if `rich-click` is not specified as a dependency in `conda-env/qd-rnaseq.yaml`.

**Comment by pbiology** [comment]
> > Did you also try to testrun this @dodslaser?
> 
> 
> 
> The included test fails if `rich-click` is not specified as a dependency in `conda-env/qd-rnaseq.yaml`.
> 
> 
> 
> 

Ok. Thought it would be installed as nf-core is a dependency, but I will add it explicitly!

**Pull request merged**